agent/data/client.go

```diff
@@ -0,0 +1,137 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package data
+
+import (
+	"path/filepath"
+	"sync"
+
+	"github.com/aws/amazon-ecs-agent/agent/api/container"
+	"github.com/aws/amazon-ecs-agent/agent/api/eni"
+	"github.com/aws/amazon-ecs-agent/agent/api/task"
+	"github.com/aws/amazon-ecs-agent/agent/engine/image"
+
+	"github.com/boltdb/bolt"
+)
+
+const (
+	dbName = "agent.db"
+	dbMode = 0600
+
+	containersBucketName     = "containers"
+	tasksBucketName          = "tasks"
+	imagesBucketName         = "images"
+	eniAttachmentsBucketName = "eniattachments"
+	metadataBucketName       = "metadata"
+)
+
+var (
+	dbClient Client
+	once     sync.Once
+)
+
+// Client specifies the data management interface to persist and manage various kinds of data in the agent.
+type Client interface {
+	// SaveContainer saves the data of a container.
+	SaveContainer(*container.Container) error
+	// SaveDockerContainer saves the data of a docker container.
+	// We have both SaveContainer and SaveDockerContainer so that a caller who doesn't have docker information
+	// of the container can save container with SaveContainer, while a caller who wants to save the docker
+	// information of the container can save it with SaveDockerContainer.
+	SaveDockerContainer(*container.DockerContainer) error
+	// DelContainer deletes the data of a container.
+	DelContainer(string) error
```

Comment:
I'd like to see these as `DeleteContainer`, `DeleteTask`, `DeleteImageState` etc.

Reply:
sure, updated